reduct-nepal/seep
# Code refractoringtominimize read operation

**Task Completed:**

- [x] code refactoring in firebase query to minimize unnecessary read operation to firebase database.
- [x] storing token in Local storage to run 'allowing user to retake the test in every 3 month' only once in a day in one browser.
- [x] allowing supeuser to view the average accuracy of the user's last test without clicking the button . 

**To test:**

- Login in Reduct Nepal Testing Portal as SuperAdmin and go to Admin Section and then view Student Record or ''/transcriptions" route
- you should be able to view the average accuracy of user last test without clicking the button.
- Every thing else work as expected with fewer read operation to firebase database
- For testing of token,inspect and go to Local storage and there you should see Token storing its expiry time.

**Expectations:**
- No quota issue while Logging in to the portal 

**Intentions:**
-To minimize the read operation to solve quota issue in firebase.
- temporary fix before moving to postgres.
